internal/service/health/health.go

```diff
@@ -2,22 +2,56 @@ package health
 
 import (
 	"context"
+	"github.com/go-kratos/kratos/v2/log"
+	"gorm.io/gorm"
 
 	pb "github.com/project-kessel/inventory-api/api/kessel/inventory/v1"
 )
 
 type HealthService struct {
 	pb.UnimplementedKesselInventoryHealthServiceServer
+	DB *gorm.DB
 }
 
-func NewHealthService() *HealthService {
-	return &HealthService{}
+func NewHealthService(db *gorm.DB) *HealthService {
+	return &HealthService{
+		DB: db,
+	}
 }
 
 func (s *HealthService) GetLivez(ctx context.Context, req *pb.GetLivezRequest) (*pb.GetLivezResponse, error) {
-	return &pb.GetLivezResponse{Status: "OK", Code: 200}, nil
+	log.Infof("Performing livez check")
+	return &pb.GetLivezResponse{
+		Status: "OK",
+		Code:   200,
+	}, nil
 }
 
 func (s *HealthService) GetReadyz(ctx context.Context, req *pb.GetReadyzRequest) (*pb.GetReadyzResponse, error) {
-	return &pb.GetReadyzResponse{Status: "OK", Code: 200}, nil
+	//determine storage type based on Gorm Dialector
+	storageType := s.DB.Dialector.Name()
+	log.Infof("Checking readiness for storage type: %s", storageType)
```

Comment:
can you move this logic to the data layer and expose via biz usecase similar to relations api